src/app/pages/things/details/things.details.component.ts

```diff
@@ -117,17 +124,25 @@ export class ThingsDetailsComponent implements OnInit, OnDestroy {
   getChannelMessages() {
     const messages: MainfluxMsg[] = [];
     this.connectedChans.forEach((chan, i) => {
-      this.messagesService.getMessages(chan.id, this.thing.key, this.thing.id).subscribe(
+      this.messagesService.getMessages(chan.id, this.thing.key, this.filters).subscribe(
         (respMsg: any) => {
-          respMsg.messages.forEach((msg: MainfluxMsg) => messages.push(msg));
-          if (i === this.connectedChans.length - 1) {
-            this.messages = messages;
+          if (respMsg.messages) {
```

Comment:
If messages is not null, i.e. if it's always an array, be it an empty array, you don't need this if. (I'm not sure if it's always non null.)

Reply:
If messages are null this field don't exist.